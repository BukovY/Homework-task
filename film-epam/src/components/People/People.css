.people_title {
    font-size: 20px;
}
.people_cover{
    width: 100%;
    height: auto;
}