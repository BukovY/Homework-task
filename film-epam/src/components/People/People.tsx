import React, { FC } from "react";
import { getPeopleCard } from "../../utils/functrions";
import { useHistory } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

type PeopleProps = {
  el: {
    id: number;
    profile_path: string;
    original_name: string;
    known_for_department: string;
  };
};

const useStyles = makeStyles(() => ({
  peopleTitle: {
    fontSize: "20px ",
  },
  peopleCover: {
    width: "100%",
    height: "auto",
    cursor: "pointer",
    borderRadius: "5px",
    aspectRatio: "0.68",
  },
  peopleImg: {
    width: "153px",
    height: "250px",
    display: "flex",
    "@media screen and (max-width: 1300px)": {
      width: "100%",
      height: "auto",
    },
  },
}));

export const People: FC<PeopleProps> = ({ el }) => {
  const classes = useStyles();
  const history = useHistory();
  const selectActor = () => {
    history.push(`/actor/${el.id}`);
  };
  return (
    <div onClick={selectActor}>
      <div className={classes.peopleImg}>
        <img
          src={getPeopleCard(el.profile_path)}
          className={classes.peopleCover}
          alt={el.original_name}
        />
      </div>
      <p className={classes.peopleTitle}>{el.original_name}</p>
      <p>{el.known_for_department}</p>
    </div>
  );
};
