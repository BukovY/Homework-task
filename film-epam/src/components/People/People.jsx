import React from "react";
import "./People.css";

const People = ({ img, title, department }) => {
  const imgPath =
    img === null
      ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
      : `https://image.tmdb.org/t/p/w500/${img}`;
  return (
    <div className="people_card">
      <img src={imgPath} className="people_cover" alt={title} />
      <p className="people_title">{title}</p>
      <p>{department}</p>
    </div>
  );
};
export default People
