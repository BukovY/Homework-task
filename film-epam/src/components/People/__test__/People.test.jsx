import React from "react";
import rerender from "react-test-renderer";
import { People } from "../People";

const el = { original_name: "Sara Connor", known_for_department: "HR" };
describe("Testing <People/> Component", () => {
  it("renders correctly", () => {
    const tree = rerender.create(<People el={el} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
