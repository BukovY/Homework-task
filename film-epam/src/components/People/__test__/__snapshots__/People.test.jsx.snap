// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <People/> Component renders correctly 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="people_img"
  >
    <img
      alt="Sara Connor"
      className="people_cover"
      src="https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    />
  </div>
  <p
    className="people_title"
  >
    Sara Connor
  </p>
  <p>
    HR
  </p>
</div>
`;
